packages/venia-ui/lib/components/_new_components/StoreInfo/storeInfo.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { useStoreInfo } from '@magento/peregrine/lib/talons/_new_talons/StoreInfo/useStoreInfo'
+import { mergeClasses } from '../../../classify';
+import defaultClasses from './storeInfo.css'
+import { shape, string } from 'prop-types'
+
+const StoreInfo = props => {
+
+    const classes = mergeClasses(defaultClasses, props.classes)
+    const talonProps = useStoreInfo({})
+
+    const { data } = talonProps;
+
+    const displayList = data ? Object.keys(data.storeConfig).map((key) => 
+    (<li className={classes.list} key={key}>{key}: {data.storeConfig[key]}</li>)) :
+    (<li className={classes.list}>No data</li>)
```

Comment:
To jest spoko ponieważ ta logika nie jest zbyt skomplikowana ale wykorzystam ten przykład do celów dydaktycznych. Jak dane przychodzą w takiej strukturze na podstawie, której ciężko jest wyrenderować kontent to należy je przerobić w komponencie logicznym używając przy tym useMemo. 
np w useStoreInfo.js
`const items = useMemo(() => {
        return data ? Object.keys(data.storeConfig).map(key => {
            return {
                key: key,
                value: data.storeConfig[key]
            }
        }) : [];
    }, [data])`
a w storeInfo.js
`items.map(({key, value}) => <li key={key}>{key}: {value} ) </li>`

useMemo jest uruchamiane tylko jeżeli zmieni się parametr podany w tablicy czyli data w tym przypadku. Dzięki czemu nie są dokonywane skomplikowane operacje za każdym razem gdy komponent się renderuje lecz tylko wtedy gdy zmienią się dane. Poza tym pozbywamy się nadmiarowej logiki z komponentu renderującego